### PR TITLE
add `selectorAll` property type, fix #1049

### LINKED ISSUE #1049: add `selectorAll` property type
Similar to `selector` property type.

Add the convenience of converting the NodeList to an array

### PR DESCRIPTION
**Pull request by legendtang**
I can't do local ci testing using latest node.js and npm. It's reported in https://github.com/aframevr/aframe/issues/1097 and might have been fixed in https://github.com/aframevr/aframe/pull/1098.

So I switched to node v0.12 as online travis does then everything goes fine.

### CODE REVIEW

**Comment by legendtang** [comment]
Redundant nodes cleared :)

**Comment by nylki** [comment]
> I can't do local ci testing using latest node.js and npm.

Doing `npm install lolex chai sinon sinon-chai` fixed it for me. Might have to be added to `package.json`.

**Comment by dmarcos** [comment]
@nylki Are you on Windows/OSX? Can you file an issue?

**Comment by nylki** [comment]
@dmarcos Fedora 23. Will file an issue.

**Comment by dmarcos** [comment]
Can you describe a use case for this? Should the ```selector``` type behave like ```selectorAll``` instead of having two separate types? @ngokevin What do you think?

**Comment by ngokevin** [comment]
It's useful when you want a list of entities (class) rather than a single entity (id). But both have distinct cases:

- selectorAll use case was to be able to pass in a class name to whitelist raycaster targets.
- selector use might be to select a single target like ```look-at```.

**Comment by ngokevin** [comment]
This looks good to me.

**Comment by dmarcos** [comment]
A single selector type that returns all the matches  can cover both cases described above: raycaster and look-at. In the case of look-at and multiple matches the component would just pick the first one.

File: tests/core/propertyTypes.test.js [hunk 1]
```diff
@@ -63,6 +63,46 @@ suite('propertyTypes', function () {
     });
   });
 
+  suite('selectorAll', function () {
+    var parse = propertyTypes.selectorAll.parse;
```
Comment (dmarcos) [inline tests/core/propertyTypes.test.js]: Thanks for the tests

**Comment by ngokevin** [comment]
It *would* work, it's better to explicitly define in the schema whether a property is intended to return a single match or multiple matches. Doesn't make sense (and is inconvenient) to have it return an array when you only want a singular match; it'd be akin to defining an int array when you intend to define an int.

File: src/core/propertyTypes.js [hunk 2]
```diff
@@ -88,6 +95,17 @@ function selectorStringify (value) {
   return defaultStringify(value);
 }
 
+function selectorAllStringify (value) {
+  if (value.item) {
+    var els = '';
+    for (var i = 0; i < value.length; ++i) {
```
Comment (dmarcos) [inline src/core/propertyTypes.js]: We can make this loop a bit cleaner:

````javascript
var i;
for (i = 0; i < value.length; ++i) { 
  els +=  '#' + value[i].getAttribute('id');
  if (i !== value.length - 1) { els += ','; }
}

**Comment by dmarcos** [comment]
On more style change and this is ready to go

File: src/core/propertyTypes.js (hunk 2 above)
Comment (ngokevin) [inline src/core/propertyTypes.js]: Filed https://github.com/aframevr/aframe/issues/1113 for later

**Comment by legendtang** [comment]
Here we go for this! We need a better way to stringify those types indeed.

**Comment by legendtang** [comment]
Close due to dev deprecated.

**Pull request closed without merging**